include/ProjectVersion.h

```diff
@@ -28,29 +29,102 @@
+		m_build(s.section( '.', 2 ).section( '-', 1 )),
+		m_compareType(c)
 	{
 	}
 
-	static int compare( const ProjectVersion & _v1,
-						const ProjectVersion & _v2 );
+	static int compare(const ProjectVersion & v1, const ProjectVersion & v2);
+
+	const int getMajor() const { return m_major; }
+	const int getMinor() const { return m_minor; }
+	const int getRelease() const { return m_release; }
+	const QString getBuild() const { return m_build; }
+	const CompareType getCompareType() const { return m_compareType; }
+	
 
+private:
+	const int m_major;
+	const int m_minor;
+	const int m_release;
+	const QString m_build;
+	const CompareType m_compareType;
 } ;
 
 
 
 
-inline bool operator<( const ProjectVersion & _v1, const char * _str )
+inline int compStr(const ProjectVersion & v1, const char * v2)
+{
+	return ProjectVersion::compare(v1, ProjectVersion(v2));
+}
+
+
+
+inline int compStr(const ProjectVersion & v1, const QString & v2)
 {
-	return ProjectVersion::compare( _v1, ProjectVersion( _str ) ) < 0;
+	return ProjectVersion::compare(v1, ProjectVersion(v2));
 }
 
 
 
+/*
+ * ProjectVersion v. char[]
+ */
+inline bool operator<(const ProjectVersion & v1, const char * v2) { return compStr(v1, v2) < 0; }
+inline bool operator>(const ProjectVersion & v1, const char * v2) { return compStr(v1, v2) > 0; }
+inline bool operator<=(const ProjectVersion & v1, const char * v2) { return compStr(v1, v2) <= 0; }
+inline bool operator>=(const ProjectVersion & v1, const char * v2) { return compStr(v1, v2) >= 0; }
+inline bool operator==(const ProjectVersion & v1, const char * v2) { return compStr(v1, v2) == 0; }
+inline bool operator!=(const ProjectVersion & v1, const char * v2) { return compStr(v1, v2) != 0; }
+
+inline bool operator<(const char * v1, const ProjectVersion & v2) { return 0 < compStr(v2, v1); }
+inline bool operator>(const char * v1, const ProjectVersion & v2) { return 0 > compStr(v2, v1); }
+inline bool operator<=(const char * v1, const ProjectVersion & v2) { return 0 <= compStr(v2, v1); }
+inline bool operator>=(const char * v1, const ProjectVersion & v2) { return 0 >= compStr(v2, v1); }
+inline bool operator==(const char * v1, const ProjectVersion & v2) { return 0 == compStr(v2, v1); }
+inline bool operator!=(const char * v1, const ProjectVersion & v2) { return 0 != compStr(v2, v1); }
+
+/*
+ * ProjectVersion v. QString
+ */
+inline bool operator<(const ProjectVersion & v1, const QString & v2) { return compStr(v1, v2) < 0; }
+inline bool operator>(const ProjectVersion & v1, const QString & v2) { return compStr(v1, v2) > 0; }
+inline bool operator<=(const ProjectVersion & v1, const QString & v2) { return compStr(v1, v2) <= 0; }
+inline bool operator>=(const ProjectVersion & v1, const QString & v2) { return compStr(v1, v2) >= 0; }
+inline bool operator==(const ProjectVersion & v1, const QString & v2) { return compStr(v1, v2) == 0; }
+inline bool operator!=(const ProjectVersion & v1, const QString & v2) { return compStr(v1, v2) != 0; }
```

Comment:
I mean ProjectVersion extends String and string can already do `>`.  Your test cases pass in a standard `char[]` or `QString` scenario (such as when we remove ProjectVersion all together and just compare two Strings).  So they are bad test cases for our library, because if C++ falls back on the underlying data type comparators, they'll pass anyway, which is exactly what we're doing when we remove the explicit overrides.

Now why and how exactly C++ decides when to implicitly override and when not to is baffling to me.  I'm still struggling to learn this language... but that said, your examples should be changed to see the effectiveness of a good compare.... i.e.

```cpp
Q_ASSERT(ProjectVersion("1.1.3", CompareType::Minor) < "1.1.10");
```